docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,74 @@
+  
+  a) the -cp / --configProvider op param
+
+  b) the edgex_configuration_provider environmental variable which overrides the op param above
+  
+- If the service is to use the local file, it reads the file associated with the profile, config filename and config file directory op params or environmental variables) and does not contact the configuration service (Consul) for any configuration information.
+- If the service is to use the configuration service, then the service will:
+  
+  1) call on Consul and check the root in Consul to see if it is already populated with configuration information provided by the service.  If so, the service will load and use the Consul provided configuration for that service.
+  2) If the root in Consul is not populated with configuration information provided by the service, the service will read the local configuration file and populate Consul with the configuration contents in that file.
+
+NOTE: as the services now self seed and the profile allows for the appropriate seeding to occur in the service itself, it will no longer be necessary to have a Docker configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.
+
+### Overrides
+Environmental variable overrides of configuration or op param values can occur in one of two ways
+- Environmental variables can override configuration values provided by Consul.  The name of the environmental variable must match the path names in Consul.  NOTE:  environment variable overrides do not get pushed into Consul.  They only override a value read from Consul on bootstrap.
+- Environmental variables can override op params.  Op params specify opreation of the service and are configuration that is not ever in Consul. 
+
+All values overriden get logged (indicating which configuration value or op param and the new value).  
+
+## Decision 
+
+This features has been implemented (with some minor changes to be done) for consideration here:  https://github.com/edgexfoundry/go-mod-bootstrap/compare/master...lenny-intel:SelfSeed2
+ 
+## Consequences 
+
+- Docker compose files will need to be changed to remove config seed
+- Config seed repo(s) all to be archived
+- The Docker configuration files and directory (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml) that are used to populate the config seed for Docker containers can be eliminated from all the services.
+- Documentation would need to reflect removal of config seed and "self seeding" process.
+- Removes any potential issue with race conditions as each service is not responsible for its own configuration.
+- Removes some confusion on the part of users as to why a service (config-seed) starts and immediately exits.
+- Minimal impact to development cycles and release schedule 
+
```

Comment:
Add 
- Docker files will need to be modified to remove setting `profile=docker`